CRM/Utils/Number.php

```diff
@@ -104,4 +104,39 @@ public static function formatUnitSize($size, $checkForPostMax = FALSE) {
     }
   }
 
+  /**
+   * Format number for display according to the current or supplied locale.
+   *
+   * Note this should not be used in conjunction with any calls to
+   * replaceCurrencySeparators as this function already does that.
+   *
+   * @param string $amount
+   * @param string $locale
+   * @param string $currency
+   * @param int $numberOfPlaces
+   *
+   * @return string
+   * @throws \Brick\Money\Exception\UnknownCurrencyException
+   */
+  public static function formatLocaleNumeric(string $amount, $locale = NULL): string {
+    $formatter = new \NumberFormatter($locale ?? CRM_Core_I18n::getLocale(), NumberFormatter::DECIMAL);
+    return self::replaceDecimalSeparators($formatter->format($amount));
```

Comment:
This does double-conversion since NumberFormatter when given a locale already converts. When I run it with a civireport it confirms this - it displays the thousands separator as a decimal point.

I haven't double-checked the unit tests - they look like they are testing a search not the display - but it looks like they are running in en_US locale - to fully test a non-US locale then at the start it would need to do `CRM_Core_I18n::singleton()->setLocale('nl_NL');` and then set it back at the end.

Reply:
Maybe instead of:

```php
$formatter = new \NumberFormatter($locale ?? CRM_Core_I18n::getLocale(), NumberFormatter::DECIMAL);
return self::replaceDecimalSeparators($formatter->format($amount));
```

it would be

```php
$formatter = new \NumberFormatter($locale ?? CRM_Core_I18n::getLocale(), \NumberFormatter::DECIMAL);
$formatter->setSymbol(\NumberFormatter::DECIMAL_SEPARATOR_SYMBOL, CRM_Core_Config::singleton()->monetaryDecimalPoint);
$formatter->setSymbol(\NumberFormatter::GROUPING_SEPARATOR_SYMBOL, CRM_Core_Config::singleton()->monetaryThousandSeparator);
$formatter->format($amount)
```

I haven't fully tested that but a quickie CLI test works.

Reply:
Thanks for your review @demeritcowboy 
I have added the methods `setSymbol` as that makes sense in CiviCRMs context.

Reply:
Thanks. So then I think `replaceDecimalSeparators()` is no longer needed here and anyway it duplicates the one in CRM_Utils_Money. Also can you squash commits?